Feat: Getting Started with Solang Guide (#6)
solana-foundation/developer-content: developers/guides/solang/getting-started.md

[Solang](https://solang.readthedocs.io/en/v0.3.1/) is a Solidity Compiler that
allows you to write Solana programs using the Solidity programming language.

If you’re an EVM developer that’s interested in leveraging the high speed and
low fees of the Solana network, then Solang is the perfect tool for you. With
Solang, you can leverage your existing knowledge of Solidity to start building
on Solana!

## Installation

In this section, we'll help you set up your development environment for Solang.
Just follow the steps outlined below:

1. **Check Your Prerequisites**: Before diving in, ensure that you have
   [Rust](https://www.rust-lang.org/tools/install) and
   [Node.js](https://nodejs.org/en) installed on your system. Windows users will
   also need to have
   [Windows Subsystem for Linux](https://solana.com/developers/guides/setup-local-development#windows-users-only)
   set up.
2. **Solana Tool Suite Installation**: Begin by installing the
   [Solana Tool Suite](https://docs.solana.com/cli/install-solana-cli-tools),
   which includes the Solana Command Line Interface (CLI) and the latest version
   of Solang.
3. **Anchor Framework Installation**: Next, install the

>    [Anchor Framework](https://www.anchor-lang.com/docs/installation). Anchor a

Reviewer: ```suggestion
   [Anchor Framework](https://www.anchor-lang.com/docs/installation). Anchor is a
```

>    widely used framework in the Solana ecosystem used to simplify the process of

Reviewer: ```suggestion
   widely used framework in the Solana ecosystem and simplifies the process of
```

   building Solana programs. With version 0.28, you can start building with
   Solang directly through Anchor.

   As of the time of this writing, please install Anchor using the following
   command for compatibility with Solang version 0.3.1:

   ```
   cargo install --git https://github.com/coral-xyz/anchor anchor-cli --locked --force
   ```

4. **Solang Extension for VSCode**: If you're a Visual Studio Code (VSCode)
   user, it's recommended to install the Solang
   [extension](https://marketplace.visualstudio.com/items?itemName=solang.solang)
   to assist with syntax highlighting. Remember to disable any active Solidity
   extensions to ensure that the Solang extension works correctly.

## Creating a new project

Once you’ve installed the Solana CLI and Anchor, you can create a new project
with the following command:

```
anchor init project_name --solidity
```

This command generates a new project with a basic Solang smart contract and a
test file that demonstrate how to interact with the contract from the client.

## Smart Contract Overview

Next, let’s go over the starter code beginning with the contract itself. Within

> your project's `./solidity` directory, you’ll find the following contract below:

Reviewer: ```suggestion
your project's `./solidity` directory, you’ll find the following contract below, which includes:
```

> The contract includes:

Reviewer: ```suggestion

```

- A `constructor` to initialize a state variable
- A `print` function to print messages to the program logs
- A `flip` function to update the state variable
- A `get` function to return the current value of the state variable

```solidity
@program_id("F1ipperKF9EfD821ZbbYjS319LXYiBmjhzkkf5a26rC")
contract starter {
    bool private value = true;

    @payer(payer)
    constructor(address payer) {
        print("Hello, World!");
    }

    /// A message that can be called on instantiated contracts.
    /// This one flips the value of the stored `bool` from `true`
    /// to `false` and vice versa.
    function flip() public {
            value = !value;
    }

    /// Simply returns the current value of our `bool`.
    function get() public view returns (bool) {
            return value;
    }
}
```

> 

Reviewer: ```suggestion
### Important Differences

```

Compared to a standard EVM smart contract, there are two important differences
you might notice:

1. The `@program_id` annotation:

   On Solana, smart contracts are referred to as “programs”. The `@program_id`
   annotation is used to specify the on-chain address of the contract.

```solidity
@program_id("F1ipperKF9EfD821ZbbYjS319LXYiBmjhzkkf5a26rC") // on-chain contract address
```

2. The `@payer` annotation:

   When storing data on-chain, a certain amount of SOL needs to be allocated to
   cover the storage costs. The `@payer` annotation specifies the user that will
   pay the SOL required to create the account for storing the state variable.

```solidity
@payer(payer) // payer for the "data account"
constructor(address payer) {
    print("Hello, World!");
}
```

> Here's an important distinction between EVM smart contracts and Solana programs:

Reviewer: ```suggestion
### Storing of state data

An important distinction between EVM smart contracts and Solana programs is how each stores "state" variables/data:
```
